src/test/librbd/test_librbd.cc

```diff
@@ -2172,7 +2172,30 @@ static void remove_full_try(rados_ioctx_t ioctx, const std::string& image_name,
 
   ASSERT_EQ(0, rbd_close(image));
 
-  // make sure we have latest map that marked the pool full
+  // If using a separate data pool, also make the 'metadata' pool full
+  char pool_name[80];
+  rados_ioctx_get_pool_name(ioctx, pool_name, sizeof(pool_name));
+  if(strcmp(pool_name, data_pool_name.c_str())) {
+    struct rados_pool_stat_t st;
+    ASSERT_EQ(0, rados_ioctx_pool_stat(ioctx, &st));
+    cmdstr = "{\"prefix\": \"osd pool set-quota\", \"pool\": \"" +
+      std::string(pool_name) + "\", \"field\": \"max_objects\", \"val\": \"" +
+      std::to_string(st.num_objects) + "\"}";
+    cmd[0] = (char *)cmdstr.c_str();
+    ASSERT_EQ(0, rados_mon_command(rados_ioctx_get_cluster(ioctx),
+                                   (const char **)cmd, 1, "", 0, nullptr, 0,
+                                   nullptr, 0));
+    int ret1 = 0;
+    std::string temp_image_name;
+    while (!ret1) {
+      temp_image_name = TestLibRBD::get_temp_image_name();
+      ret1 = create_image(ioctx, temp_image_name.c_str(), size, &order);
+      sleep(1);
```

Comment:
Wouldn't this loop structure unnecessarily delay the test by 1 second?  Also, `ret1` variable seems redundant as we already have `ret` and it would be nice to limit the number of retries just in case.  How about something like:
```
    for (int i = 0; i < 50; i++) {
      auto temp_image_name = TestLibRBD::get_temp_image_name();
      ret = create_image(ioctx, temp_image_name.c_str(), size, &order);
      if (ret < 0) {
        break;
      }
      sleep(1);
    }
    ASSERT_EQ(ret, -EDQUOT);
```